neo4j-cypher-dsl-parser/src/test/java/org/neo4j/cypherdsl/parser/ParserIssuesIT.java

```diff
@@ -0,0 +1,339 @@
+		var normalized = renderer.render(CypherParser.parse(cypher));
+
+		assertThat(normalized).isEqualTo("""
+			MATCH (v0:Actor)
+			CALL {
+			  WITH v0
+			  CALL {
+			    WITH v0
+			    MATCH (v0)-[v1:ACTED_IN]->(v2:Movie)
+			    WITH {
+			      properties: {
+			        screenTime: v1.screenTime
+			      },
+			      node: {
+			        __resolveType: 'Movie',
+			        title: v2.title
+			      }
+			    } AS v3
+			    RETURN v3 UNION
+			    WITH v0
+			    MATCH (v0)-[v4:ACTED_IN]->(v5:Series)
+			    WITH {
+			      properties: {
+			        screenTime: v4.screenTime
+			      },
+			      node: {
+			        __resolveType: 'Series',
+			        episodes: v5.episodes,
+			        title: v5.title
+			      }
+			    } AS v3
+			    RETURN v3
+			  }
+			  WITH collect(v3) AS v1
+			  RETURN {
+			    edges: v1
+			  } AS v2
+			}
+			RETURN v0 {
+			  .name,
+			  actedInConnection: v2
+			} AS v0""");
+	}
+
+	@Test
+	void projectionsShouldBeSortedToo() {
+		var cypher = """
+			MATCH (user:User)
+			return user {
+			   .title,
+			   b: 'B',
+			   .*,
+			   a: 'A'
+			 } AS user
+			""";
+
+		Statement statement = CypherParser.parse(cypher, Options.newOptions()
+			.createSortedMaps(true) // this has no effect on the projection
+			.build());
+
+		var renderer = Renderer.getRenderer(Configuration.newConfig()
+			.withPrettyPrint(true)
+			.withGeneratedNames(true)
+			.build());
+		var normalized = renderer.render(statement);
+		assertThat(normalized).isEqualTo("""
+			MATCH (v0:User)
+			RETURN v0 {
+			  a: 'A',
+			  b: 'B',
+			  .*,
+			  .title
+			} AS v0""");
+	}
+
+	@Test
+	void aliasesShouldBeReusedToo() {
+		var q1 = """
+			MATCH (this:Film:Multimedia)
+			CALL {
+				WITH this
+				MATCH (this_actors:Actor:Person)-[this0:ACTED_IN]->(this)
+				WITH this_actors {
+					.name
+				} AS this_actors
+				RETURN collect(this_actors) AS this_actors
+			}
+			RETURN this {
+				.title,
+				actors: this_actors
+			} AS this
+			""";
+
+		var q2 = """
+			MATCH (this:Film:Multimedia)
+			CALL {
+				WITH this
+				MATCH (actor0:Actor:Person)-[actedIn0:ACTED_IN]->(this)
+				WITH actor0 {
+					.name
+				} AS actors
+				RETURN collect(actors) AS actors
+			}
+			RETURN this {
+				.title,
+				actors: actors
+			} AS this
+			""";
+
+		var parserConfig = Options.newOptions()
+			.createSortedMaps(true)
+			.alwaysCreateRelationshipsLTR(true)
+			.build();
+		var renderer = Renderer.getRenderer(Configuration.newConfig()
+			.withPrettyPrint(true)
+			.withGeneratedNames(true)
+			.build());
+
+		var s1 = renderer.render(CypherParser.parse(q1, parserConfig));
+		var s2 = renderer.render(CypherParser.parse(q2, parserConfig));
+
+		var expected1 = """
+			MATCH (v0:Film:Multimedia)
+			CALL {
+			  WITH v0
+			  MATCH (v1:Actor:Person)-[v2:ACTED_IN]->(v0)
+			  WITH v1 {
+			    .name
+			  } AS v1
+			  RETURN collect(v1) AS v1
+			}
+			RETURN v0 {
+			  actors: v1,
+			  .title
+			} AS v0""";
+
+		var expected2 = """
```

Comment:
I think we should make this behavior configurable. For my use case (semantic comparison), the reuse of aliases is more of a hindrance than a benefit, since the two normalized queries are still not identical and I even have more queries that no longer match due to this adjustment.
Background: in neo4j-graphql-java I potentially use different aliases than those used in neo4j-grapqhl.

Reply:
I think I’ll make this opt out then, as reusing the alias fits the original query more.

Reply:
Added one additional enum constant to remove this behaviour, see the test I added for usage. Good suggestion, thanks.